include/forcing/NullForcingProvider.hpp

```diff
@@ -0,0 +1,271 @@
+#ifndef NGEN_NULLFORCING_H
+#define NGEN_NULLFORCING_H
+
+#include <vector>
+#include <set>
+#include <cmath>
+#include <algorithm>
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <unordered_map>
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+#include "CSV_Reader.h"
+#include <ctime>
+#include <time.h>
+#include <memory>
+#include "AorcForcing.hpp"
+#include "GenericDataProvider.hpp"
+#include "DataProviderSelectors.hpp"
+#include <exception>
+#include <UnitsHelper.hpp>
+
+/**
+ * @brief Forcing class providing time-series precipiation forcing data to the model.
+ */
+class NullForcingProvider : public data_access::GenericDataProvider
+{
+    public:
+
+    typedef struct tm time_type;
+
+
+    NullForcingProvider(forcing_params forcing_config):start_date_time_epoch(forcing_config.simulation_start_t),
+                                           end_date_time_epoch(forcing_config.simulation_end_t),
+                                           current_date_time_epoch(forcing_config.simulation_start_t),
+                                           forcing_vector_index(-1)
+    {
+        read_csv();
+    }
+
+    // BEGIN DataProvider interface methods
+
+    /**
+     * @brief the inclusive beginning of the period of time over which this instance can provide data for this forcing.
+     *
+     * @return The inclusive beginning of the period of time over which this instance can provide this data.
+     */
+    long get_data_start_time() override {
+        //FIXME: Trace this back and you will find that it is the simulation start time, not having anything to do with the forcing at all.
```

Comment:
Ugh. I am reminded of this. It may not matter in the case of NullForcingProvider... it also may only matter for legacy formulations... we'll have to see. No action here yet, lets see what it does with the values you have.

Reply:
It seems to be the only function in NullForcingProvider that appears to be of some use.

Reply:
So, it seems that this is fine. It turns out--I think--that this needs to be simulation time so that when a request comes _back to the forcing object_ the calculated time or index in the request will be correct... but in the case of NullForcingProvider, no requests for data are ever made back to it, so it never becomes a problem.

This still leaves open the question of how a BMI-based "forcing" provider will handle time offsets with real time values, but that's definitely a separate issue.

Reply:
I have to go back to look at it. It seems so far time offsets has not been used so far. One option would be to let the forcing engine to handle this.